morango/sync/backends/postgres.py

```diff
@@ -268,3 +277,63 @@ def _dequeuing_insert_remaining_rmcb(self, cursor, transfersession_id):
         )
 
         cursor.execute(insert_remaining_rmcb)
+
+    def _execute_lock(self, key1, key2=None, unlock=False, session=False, shared=False, wait=True):
+        """
+        Creates or destroys an advisory lock within postgres
+        :param key1: An int sent to the PG lock function
+        :param key2: A 2nd int sent to the PG lock function
+        :param unlock: A bool representing whether query should use `unlock`
+        :param session: A bool indicating if this should persist outside of transaction
+        :param shared: A bool indicating if this should be shared, otherwise exclusive
+        :param wait: A bool indicating if it should use a `try` PG function
+        """
+        if not session:
+            if not self.connection.in_atomic_block:
+                raise NotImplementedError("Advisory lock requires transaction")
+            if unlock:
+                raise NotImplementedError("Transaction level locks unlock automatically")
+
+        keys = [key1]
+        if key2 is not None:
+            keys.append(key2)
+
+        query = "SELECT pg{_try}_advisory_{xact_}{lock}{_shared}({keys}) AS lock;".format(
+            _try="" if wait else "_try",
+            xact_="" if session else "xact_",
+            lock="unlock" if unlock else "lock",
+            _shared="_shared" if shared else "",
+            keys=", ".join(["%s" for i in range(0, 2 if key2 is not None else 1)])
```

Comment:
Took me a moment to parse the list comprehension here. Totally optional, but one alternative format could be something like `["%s"] * len(keys)`.